### PR TITLE
feat: migrate StoryState.ts

### PR DESCRIPTION
**Pull request by ephread**
Finally ready for review! I don't have much to say about this one, it's the usual thing.

### CODE REVIEW

**Comment by ephread** [comment]
Up! @y-lohse, I hope you'll get a chance to look at this chunky PR soon.

File: src/engine/StoryState.ts [hunk 1]
```diff
@@ -981,11 +1041,13 @@ export class StoryState {
...
+  public PassArgumentsToEvaluationStack(args: any[] | null) {
+    if (args !== null) {
       for (let i = 0; i < args.length; i++) {
-        if (!(typeof args[i] === "number" || typeof args[i] === "string")) {
+        if (
+          !(typeof args[i] === "number" || typeof args[i] === "string") ||
+          args[i] instanceof InkList
+        ) {
           throw new Error(
             "ink arguments when calling EvaluateFunction / ChoosePathStringWithParameters  must be int, float or string"
```
Comment (y-lohse) [inline src/engine/StoryState.ts:1052]: The error message has also changed upstream, on l.1153 :)

File: src/engine/StoryState.ts [hunk 2]
```diff
@@ -536,23 +571,68 @@ export class StoryState {
       );
     }
 
-    this.callStack.SetJsonToken(jObject["callstackThreads"], this.story);
+    let flowsObj = jObject["flows"];
+    if (flowsObj != null) {
+      let flowsObjDict = flowsObj as Record<string, any>;
+
+      // Single default flow
+      if (Object.keys(flowsObjDict).length == 1) {
```
Comment (y-lohse) [inline src/engine/StoryState.ts:579]: There are 2 double equal compares on lengths around here, I think they could be strict equals like everywhere else?

**Review by y-lohse — APPROVED** [review]
I apologize for the delay in review 😭. If this happens again, feel free to merge things without my approval if you're comfortable with that, it's insane how long it took me to review this.

I left 2 small comments but none is a blocker :)

**Comment by ephread** [comment]
> I apologize for the delay in review 😭.

No worries!

> If this happens again, feel free to merge things without my approval if you're comfortable with that, it's insane how long it took me to review this.

Got it, but you did find some mistakes in there! It's all fixed now, so I'm merging.

**Pull request merged**